frontend/server/libs/dao/Problemsets.dao.php

```diff
@@ -39,30 +39,35 @@ public static function getProblemsetContainer($problemset_id) {
      *  Check if a submission is before the deadline.
      *  No one, including admins, can submit after the deadline.
      */
-    public static function isLateSubmission($container) {
+    public static function isLateSubmission(Object $container) : bool {
         return isset($container->finish_time) &&
                (Time::get() > strtotime($container->finish_time));
     }
 
-    public static function insideSubmissionWindow($container, $identity_id) {
+    public static function insideSubmissionWindow(
+        Object $container,
+        int $identityId,
+        ?ProblemsetIdentities $problemsetIdentities = null
+    ) : bool {
         if (isset($container->finish_time)) {
             if (Time::get() > strtotime($container->finish_time) ||
                 Time::get() < strtotime($container->start_time)) {
-                return false;
+                if (is_null($problemsetIdentities)) {
+                    return false;
+                }
+                return Time::get() <= strtotime($problemsetIdentities->end_time);
```

Comment:
```suggestion
                return !is_null($problemsetIdentities) && Time::get() <= strtotime($problemsetIdentities->end_time);
```

así puedes borrar L55-L57.

también, puedes agregar una prueba donde un concursante intente hacer un envío antes de `start_time`? según entiendo, con este cambio, el usuario podría hacerlo!

Reply:
Listo, la prueba ya existía en `testRunWhenContestNotStarted`, pero estaba entrando a una pre-condición que no permitía hacer el envío. Creo que así es como debe quedar